src/screencast/screencast.c

```diff
@@ -164,7 +169,7 @@ bool setup_target(struct xdpw_screencast_context *ctx, struct xdpw_session *sess
 		free(target);
 		return false;
 	}
-	assert(target->output);
+	assert(target->output || target->handle);
```

Comment:
This accesses members of a union before checking the `type` field.

TBH, I think it'd be better to drop the union here to avoid these footguns.